assets/src/components/amp-support/index.js

```diff
@@ -0,0 +1,176 @@
+import { SiteInfo } from './site-info';
+import { Theme } from './theme';
+import { Plugins } from './plugins';
+import { Errors } from './errors';
+import { ErrorSources } from './error-sources';
+import { ValidatedUrls } from './validated-urls';
+import { RawData } from './raw-data';
+
+/**
+ * AMP Support component.
+ *
+ * @class
+ * @param {Object} props Props for component.
+ * @return {JSX.Element} Makrup for AMP support component
+ */
+export function AMPSupport( props ) {
+	const { data, restEndpoint, args, nonce } = props;
+
+	const [ sending, setSending ] = useState( false );
+	const [ uuid, setUuid ] = useState( null );
+	const [ error, setError ] = useState( null );
+	const [ hasCopied, setHasCopied ] = useState( false );
+	const [ submitSupportRequest, setSubmitSupportRequest ] = useState( false );
+
+	/**
+	 * Event callback for send button.
+	 */
+	useEffect( () => {
+		( async () => {
+			if ( false === submitSupportRequest ) {
+				return;
+			}
+
+			setSending( true );
+			setUuid( null );
+			setError( null );
+
+			try {
+				apiFetch.use( apiFetch.createNonceMiddleware( nonce ) );
+
+				const response = await apiFetch(
+					{
+						path: restEndpoint,
+						method: 'POST',
+						data: {
+							args,
+						},
+					},
+				);
+
+				if ( undefined !== response.success && undefined !== response.data ) {
+					setUuid( response.data.uuid );
+				} else {
+					setSubmitSupportRequest( false );
+					throw new Error( __( 'Failed to send support request. Please try again after some time', 'amp' ) );
+				}
+			} catch ( exception ) {
+				setError( exception.message );
+			} finally {
+				setSending( false );
+			}
+		} )();
+	}, [ submitSupportRequest, restEndpoint, args, nonce ] );
+
+	return (
+		<div className="amp-support">
+			<Selectable>
+				<h2 className="amp-support__heading">
+					{ __( 'AMP Support', 'amp' ) }
+				</h2>
+				{ /* dangerouslySetInnerHTML reason: Injection of links. */ }
+				<p dangerouslySetInnerHTML={
+					{
+						__html: sprintf(
+							/* translators: %s is the URL to create a new support topic */
+							__( 'In order to best assist you, please submit the following information to our private database. Once you have done so, copy the the resulting support ID and mention it in a <a href="%s" rel="noreferrer" target="_blank">support forum topic</a>. You do not have to submit data to get support, but our team will be able to help you more effectively if you do so.', 'amp' ),
+							'https://wordpress.org/support/plugin/amp/#new-topic-0',
+						),
+					}
+				} />
+				<div className="amp-support__body">
+
+					{ data.site_info && <SiteInfo data={ data.site_info } /> }
+
+					{ data.themes && <Theme data={ data.themes } /> }
+
+					{ data.plugins && <Plugins data={ data.plugins } /> }
+
+					{ data.errors && <Errors data={ data.errors } /> }
+
+					{ data.error_sources && <ErrorSources data={ data.error_sources } /> }
+
+					{ data.urls && <ValidatedUrls data={ data.urls } /> }
```

Comment:
Can we rename the generic `data` props to a clearer and more verbose names like:

```suggestion
					{ data.site_info && <SiteInfo siteInfo={ data.site_info } /> }

					{ data.themes && <Theme themes={ data.themes } /> }

					{ data.plugins && <Plugins plugins={ data.plugins } /> }

					{ data.errors && <Errors errors={ data.errors } /> }

					{ data.error_sources && <ErrorSources errorSources={ data.error_sources } /> }

					{ data.urls && <ValidatedUrls urls={ data.urls } /> }
```

Reply:
I have updated the props name.